Updates to federation docs for Rover launch (#731)
apollographql/federation: docs/shared/register-federated-cli.mdx

> In a federated architecture, each of your graph's [subgraphs](https://www.apollographql.com/docs/federation/implementing-services/) uses the Rover CLI to register its schema with Apollo:

Reviewer: Any chance we could change the URL to reflect "subgraphs" too?

Author: This will be in a big unwieldy followup PR that just replaces "implementing service" with "subgraph" all over the place

```mermaid
graph LR;
  subgraph " "
  gateway([Gateway]);
  serviceA[Products subgraph];
  serviceB[Reviews subgraph];
  serviceC[Inventory subgraph];
  end
  registry{{Apollo Schema Registry}};
  gateway --- serviceA & serviceB & serviceC;

>   serviceA & serviceB & serviceC -. rover subgraph publish  .-> registry;

Author: These are effectively variable names for rendering the SVG diagram on the page. I'll update them to `subgraphA` etc for nomenclature consistency, but easier to reason about them as a group in the "code" this way. Here's what it looks like rendered on the page:

<img width="742" alt="Screen Shot 2021-05-06 at 3 20 48 PM" src="https://user-images.githubusercontent.com/3433000/117372698-c8a06800-ae7e-11eb-83a1-3630ebf25d87.png">

  class registry secondary;
```

If you haven't yet:

*  <a href="https://www.apollographql.com/docs/rover/getting-started/" target="_blank">Install the Rover CLI.</a>
* <a href="https://www.apollographql.com/docs/rover/configuring/" target="_blank">Authenticate Rover</a> with Apollo Studio.

Then, **do the following for each of your subgraphs**:

1. Obtain the following values, which are required for the `rover subgraph publish` command:

>     * The name that uniquely identifies the subgraph within your graph (e.g., `products`).

Reviewer: We may want to be more clear about where to find this -- and help people understand that they should not use a name already in use (this is an area where our system is more forgiving that we'd probably like, so giving people some guidance on this will be important).

    * The URL that your gateway will use to communicate with the subgraph (e.g., `http://products-graphql.svc.cluster.local:4001/`).
    * The service's schema. The Rover CLI can either:
        * Use a `.gql` or `.graphql` file saved on your local machine, or
        * Perform an introspection query on the running service to fetch the schema

2. Run the `rover subgraph publish` command, providing your schema in one of the ways shown:

    ```bash
    # Provide a local .graphql file path
    rover subgraph publish my-graph@my-variant --name products --routing-url http://products-graphql.svc.cluster.local:4001/ --schema ./schema.graphql 

    # Provide an introspection result via stdin

>     rover graph introspect http://localhost:4000 | rover subgraph publish my-graph@my-variant --name products --routing-url http://products-graphql.svc.cluster.local:4001/ --schema -

Reviewer: We should consider using `subgraph introspect` here, which is a more common case and yields the "service SDL" with federation directives instead of a plain introspection result.

    ```

> As you register your subgraph schemas, the schema registry attempts to **compose** their latest versions into a single **supergraph schema**. Whenever composition succeeds, your gateway can fetch the latest supergraph schema from the registry.

Reviewer: Coming soon is `rover supergraph fetch`, which we _might_ want to mention here. We can also tell them they can then view their supergraph in the Schema->SDL tab in Studio.

Author: Will this command be included in 1.0?

Reviewer 2: it'll be in the next release we cut! i think we're going for 0.1.0 instead of 1.0.0 but, yes!

  serviceA & serviceB & serviceC -. rover subgraph publish  .-> registry;
  class registry secondary;
```

If you haven't yet:

*  <a href="https://www.apollographql.com/docs/rover/getting-started/" target="_blank">Install the Rover CLI.</a>
* <a href="https://www.apollographql.com/docs/rover/configuring/" target="_blank">Authenticate Rover</a> with Apollo Studio.

Then, **do the following for each of your subgraphs**:

1. Obtain the following values, which are required for the `rover subgraph publish` command:

    * The name that uniquely identifies the subgraph within your graph (e.g., `products`).
        * If you've already registered a particular subgraph at least once, you can get its name from your graph's **Schema > SDL** tab in Apollo Studio, or by running `rover subgraph list`.
        * Otherwise, you set the subgraph's name the first time you register it. **Make sure the name is unique within your graph.**
    * The URL that your gateway will use to communicate with the subgraph (e.g., `http://products-graphql.svc.cluster.local:4001/`).
    * The service's schema. The Rover CLI can either:
        * Use a `.gql` or `.graphql` file saved on your local machine, or

>         * Perform an introspection query on the running service to fetch the schema

Reviewer 2: I wonder if we should just remove these two from step 1 since we explain it in more detail in step 2. I imagine folks might come across this reading linearly and not realize they can move on to step 2 before actually getting the schema.

2. Run the `rover subgraph publish` command, providing your schema in one of the ways shown:

    ```bash
    # Provide a local .graphql file path
    rover subgraph publish my-graph@my-variant --name products --routing-url http://products-graphql.svc.cluster.local:4001/ --schema ./schema.graphql 

    # Provide an introspection result via stdin
    rover subgraph introspect http://localhost:4000 | rover subgraph publish my-graph@my-variant --name products --routing-url http://products-graphql.svc.cluster.local:4001/ --schema -
    ```

As you register your subgraph schemas, the schema registry attempts to **compose** their latest versions into a single **supergraph schema**. Whenever composition succeeds, your gateway can fetch the latest supergraph schema from the registry.

> You can also fetch your latest supergraph schema with the `rover supergraph fetch` command, and it's available in Apollo Studio from your graph's **Schema > SDL** tab.

Reviewer 2: nit: just kinda like this wording better. feel free to disregard

```suggestion
You can also manually fetch your latest supergraph schema with the `rover supergraph fetch` command, or retrieve it from your graph's **Schema > SDL** tab in Apollo Studio.
```
